### PR TITLE
[feature](schema change) unified schema change for parquet and orc reader (#32873)

### PR DESCRIPTION
**Pull request by hubgeter**
## Proposed changes
bp #32873 

Scenario: Reading a hive table after adding fields to a struct column
Since there are still problems with reading tables in parquet and text formats on the master in this scenario, only tables in orc format are picked here and some cases are added.

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

File: be/src/vec/data_types/data_type_struct.cpp [hunk 1]
```diff
@@ -355,14 +355,7 @@ std::optional<size_t> DataTypeStruct::try_get_position_by_name(const String& nam
...
 String DataTypeStruct::get_name_by_position(size_t i) const {
-    if (i == 0 || i > names.size()) {
-        fmt::memory_buffer error_msg;
-        fmt::format_to(error_msg, "Index of tuple element ({}) if out range ([1, {}])", i,
-                       names.size());
-        LOG(FATAL) << fmt::to_string(error_msg);
-    }
-
-    return names[i - 1];
+    return names[i];
```
Comment (morningman) [inline be/src/vec/data_types/data_type_struct.cpp:358]: Why removing this check?

Comment (hubgeter) [reply be/src/vec/data_types/data_type_struct.cpp:358]: because : https://github.com/apache/doris/pull/32873/files#r1555136892

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**